servers/rendering/renderer_rd/shaders/scene_forward_clustered.glsl

```diff
@@ -1378,62 +1378,78 @@ void main() {
 				} else { //no soft shadows
 
 					vec4 pssm_coord;
+					float blur_factor;
+
 					if (depth_z < directional_lights.data[i].shadow_split_offsets.x) {
 						vec4 v = vec4(vertex, 1.0);
 
 						BIAS_FUNC(v, 0)
 
 						pssm_coord = (directional_lights.data[i].shadow_matrix1 * v);
+						blur_factor = 1.0;
 					} else if (depth_z < directional_lights.data[i].shadow_split_offsets.y) {
 						vec4 v = vec4(vertex, 1.0);
 
 						BIAS_FUNC(v, 1)
 
 						pssm_coord = (directional_lights.data[i].shadow_matrix2 * v);
+						// Multiply shadow blur by the 3rd split's share to reduce discrepancy between shadow splits.
+						blur_factor = directional_lights.data[i].shadow_split_offsets.z / directional_lights.data[i].shadow_split_offsets.w;
```

Comment:
Why is the blur for the second shadow split calculated based on the 3rd split?

Shouldn't this be based on the relative size of the split itself?

Reply:
I don't really know why it's working this way, but it seems to be working correctly…

Reply:
I have a feeling that it is working by chance and that different settings won't produce the same nice results. We should come up with a more principled factor to reduce the amount of blur.

Reply:
The assumption of each split having half the resolution of the previous split is false, for a few reasons:
1) Each split has same Shadow map Resolution assigned to them (1/Split count of the total Atlas)
2) if Split 1 is 10% of the distance, and Split 2 is 30% of the max distance, the 2nd split isn't half the effective resolution of Split 1, as it's rendering 30%, including the 10% of Split 1, making it 1/3 of the effective resolution of Split 1.

Do keep in mind, the 10%,20%,50% splits are only the defaults.

Now that I think of it, there is possibly an easy way to Determine split blur probably...
When looking at the Default of 1 , 0.5 , 0.2 , 0.1 I notice a pattern. Default splits are 0.1 , 0.2 , 0.5 , 1
See where I'm going?
They're just in reverse order, which makes sense, as the effective shadow resolution is proportional to their size, and blur is connected to effective resolution so...

Reply:
I should try playing around with split values in DirectionalLight3D and see if blurring still looks good.

Reply:
Right, so we are back to calculating the effective area based on split size. I guess we have two options then. We can adjust the blur amount with reference to the entire atlas, or with reference to the first split. 

I think adjusting with reference to the first split makes more sense (which is what this PR already does).

So, what we need to do is calculate the relative sizes one way of doing that is:
``shadow_split_offset.x / shadow_split_offset[i]``
With default values you get the same results as what you have (first split can still be hardcoded to 1.0). But with this, when you move away from default values the blur scale is determined with reference to the relative split size between the first split and the current split. Instead of being determined with reference to 1.0 and a different split.
*Before*
| split | size | blur_factor |
|-------|------|-------------|
| 1     | 0.1  | 1.0         |
| 2     | 0.2  | 0.5         |
| 3     | 0.5  | 0.2         |
| 4     |      | 0.1         |

| split | size | blur_factor |
|-------|------|-------------|
| 1     | 0.19  | 1.0         |
| 2     | 0.2  | 0.5         |
| 3     | 0.5  | 0.2         |
| 4     |      | 0.19         |

*After*
| split | size | blur_factor |
|-------|------|-------------|
| 1     | 0.1  | 1.0         |
| 2     | 0.2  | 0.5         |
| 3     | 0.5  | 0.2         |
| 4     |      | 0.1         |

| split | size | blur_factor |
|-------|------|-------------|
| 1     | 0.19  | 1.0         |
| 2     | 0.2  | 0.95         |
| 3     | 0.5  | 0.2         |
| 4     |      | 0.1         |